types/gavazzi/UWPA_UWPM/uplink.js

```diff
@@ -0,0 +1,66 @@
+// INT32 - Little Endian
+function INT32LE(hex, signed) {
+  const data = hex.match(/../g);
+  const buf = new ArrayBuffer(4);
+  const view = new DataView(buf);
+  let num = 0;
+
+  if (signed) {
+    data.forEach((b, i) => {
+      view.setInt8(i, parseInt(b, 16));
+    });
+    num = view.getInt32(0, 1);
+  } else {
+    data.forEach((b, i) => {
+      view.setUint8(i, parseInt(b, 16));
+    });
+    num = view.getUint32(0, 1);
+  }
+
+  return num;
+}
+
+function consume(event) {
+  const payload = event.data.payloadHex;
+  const bits = Bits.hexToBits(payload);
+  const data = {};
+
+  const appContext = Bits.bitsToUnsigned(bits.substr(0, 8));
+  let pointer = 2;
+
+  if (appContext === 1) {
```

Comment:
what are the other modes?

Reply:
Probably different phases, but we came to the conclusion that we (for now) only use our config.